packages/webapp/src/containers/LocationDetails/LocationIrrigation/useIrrigationPrescriptions.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ *  Copyright 2025 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useGetIrrigationPrescriptionsQuery } from '../../../store/api/apiSlice';
+import { Location, Task } from '../../../types';
+import { IrrigationPrescription } from '../../../store/api/types';
+import { useSelector } from 'react-redux';
+import { tasksSelector } from '../../taskSlice';
+import { getTaskStatus } from '../../Task/taskCardContentSelector';
+
+interface LocationIrrigationPrescription extends IrrigationPrescription {
+  task?: Task;
+}
+
+const ONE_HOUR_IN_MS = 1000 * 60 * 60;
+const getMockData = (location: Location, tasks: Task[]): IrrigationPrescription[] => [
+  {
+    id: 'uuid_maybe_001',
+    some_location_id: location.location_id,
+    prescription_date: new Date(Date.now() - ONE_HOUR_IN_MS).toDateString(),
+    partner_id: 1,
+    task_id: tasks.length ? tasks.at(-1)?.task_id : undefined,
+  },
+  {
+    id: 'uuid_maybe_002',
+    some_location_id: location.location_id,
+    prescription_date: new Date(Date.now() - ONE_HOUR_IN_MS).toDateString(),
+    partner_id: 1,
+    task_id: undefined,
+  },
+];
+
+export default function useIrrigationPrescriptions(location?: Location) {
+  if (!location) {
+    return [];
+  }
+
+  const { data, error, isFetching } = useGetIrrigationPrescriptionsQuery();
```

Comment:
The query will actually need to be called (or pre-fetched) as soon as the app loads, since the GET endpoint for IPs will be taking care of generating the notification if there's a new IP -- this should still happen even if the user doesn't specifically navigate to this view.

Reply:
I can add it there too if you think I should, I might be now confused about what this PR should cover and what 
Ticket LF # 4762 (avoiding linking for jira auto) should cover.

Reply:
The notification ticket covers modifying the backend method to check if the notification exists and if not, create one, but the call to the endpoint itself when the app loads from the UI belongs in the ticket related to this PR according to the first item

> Fetch IPs by calling GET /irrigation_prescriptions when app loads.
> If 24 hs have elapsed since last prescription timestamp, IPs should be re-fetched.

So at this point there shouldn't be any notifications happening, but the endpoint call should be happening at the right time.

Reply:
I think this is resolved that we will be instead u sing the existing notification subscription and handling on BE. So use of the data where it is needed is good.